src/InventoryList/Inventory.pm

```diff
@@ -0,0 +1,49 @@
+#########################################################################
+#  OpenKore - Player actor object
+#  Copyright (c) 2005 OpenKore Team
+#
+#	TEST BY HENRYBK
+#
+#########################################################################
+package InventoryList::Inventory;
+
+use strict;
+use Globals;
+use InventoryList;
+use base qw(InventoryList);
+
+use constant {
+	MAP_LOADED => 0,
+	RECV_STAT_INFO2 => 1
+};
+
+sub new {
+	my ($class) = @_;
+	my $self = $class->SUPER::new('Inventory');
+	$self->{hooks} = Plugins::addHooks (
```

Comment:
Do these hooks ever get removed? It's not safe to assume `DESTROY` will remove them either, because the garbage collector may not be run immediately when the object is freed. And because the hook is a closure, this object is a memory leak. Does it make sense to assume that only one inventory can ever exist? If so, the hook could be a global and this object could hide the global state. If we don't want to do that (and I'd love to see a day where a single instance of OpenKore could manage multiple characters), then we need logic to remove these hooks when we allocate a new `$char` object.